libclc/libspirv/lib/amdgcn-amdhsa/misc/sub_group_shuffle.cl

```diff
@@ -980,10 +988,10 @@ __AMDGCN_CLC_SUBGROUP_DOWN_TO_VEC(ulong8, m, 8)
 __AMDGCN_CLC_SUBGROUP_DOWN_TO_VEC(ulong16, m, 16)
 // half
 #ifdef cl_khr_fp16
-__AMDGCN_CLC_SUBGROUP_DOWN_TO_VEC(half2, DF16_, 2)
-__AMDGCN_CLC_SUBGROUP_DOWN_TO_VEC(half4, DF16_, 4)
-__AMDGCN_CLC_SUBGROUP_DOWN_TO_VEC(half8, DF16_, 8)
-__AMDGCN_CLC_SUBGROUP_DOWN_TO_VEC(half16, DF16_, 16)
+__AMDGCN_CLC_SUBGROUP_DOWN_TO_VEC(clc_fp16x2_t, DF16_, 2)
```

Comment:
So we still want this mangling, correct?

Reply:
Aye we do. We're still doing manual mangling here because (for some reason) we're generating builtins we can't express in OpenCL C, e.g., `_Float16 __spirv_SubgroupShuffleINTEL<_Float16>(_Float16, unsigned int)`.

I'd like to get rid of them, but I don't think we can remove builtins as it constitutes an ABI break (https://github.com/intel/llvm/pull/17732#discussion_r2021343904).